test/TritonGPU/combine.mlir

```diff
@@ -2060,6 +2060,22 @@ module attributes {"triton_gpu.compute-capability" = 80 : i32, "triton_gpu.num-c
     }) : (tensor<1x32xf64, #blocked1>, tensor<1x32xi32, #blocked1>) -> (tensor<1xf64, #triton_gpu.slice<{dim = 1, parent = #blocked1}>>, tensor<1xi32, #triton_gpu.slice<{dim = 1, parent = #blocked1}>>)
     %4 = triton_gpu.convert_layout %3#1 : (tensor<1xi32, #triton_gpu.slice<{dim = 1, parent = #blocked1}>>) -> tensor<1xi32, #triton_gpu.slice<{dim = 1, parent = #blocked}>>
     tt.return %4 : tensor<1xi32, #triton_gpu.slice<{dim = 1, parent = #blocked}>>
+}
+}  // end module
+
+// -----
+
+#blocked = #triton_gpu.blocked<{sizePerThread = [1,2], threadsPerWarp = [32,1], warpsPerCTA = [1,1], order = [1,0]}>
+#blocked1 = #triton_gpu.blocked<{sizePerThread = [1,1], threadsPerWarp = [16,2], warpsPerCTA = [1,1], order = [1,0]}>
+#blocked2 = #triton_gpu.blocked<{sizePerThread = [1], threadsPerWarp = [32], warpsPerCTA = [1], order = [0]}>
+#blocked3 = #triton_gpu.blocked<{sizePerThread = [2], threadsPerWarp = [32], warpsPerCTA = [1], order = [0]}>
+
+module attributes {"triton_gpu.num-ctas" = 1 : i32, "triton_gpu.num-warps" = 1 : i32, "triton_gpu.threads-per-warp" = 32 : i32} {
+  tt.func public @transpose_reshape(%arg0: tensor<16x2xf32, #blocked>) -> tensor<32xf32, #blocked3> {
+    %a = triton_gpu.convert_layout %arg0 : (tensor<16x2xf32, #blocked>) -> tensor<16x2xf32, #blocked1>
+    %b = tt.reshape %a {allow_reorder = false} : tensor<16x2xf32, #blocked1> -> tensor<32xf32, #blocked2>
+    %c = triton_gpu.convert_layout %b : (tensor<32xf32, #blocked2>) -> tensor<32xf32, #blocked3>
+    tt.return %c : tensor<32xf32, #blocked3>
   }
```

Comment:
Would be nice to also test the forward propagation on its own with something like:
```
convert
reshape
ret
```

Reply:
Actually I'm not sure we can meaningfully do this.

Because reshape is a nop, if you want to have a testcase that shows that the `convert` goes away, either `convert` itself is a nop (which ofc you don't want, that can be eliminated in other ways), or you need two converts which cancel each other out, on either side of the reshape.

Reply:
right, I meant writing something like:
convert
reshape

and checking that we end up with
reshape
convert

since our current algorithm sinks the convert, otherwise we could also add an op which folds the convert in like `cat` or `histogram`

Reply:
OK, interestingly when I make this work, I break the A100 tests because I'm adding additional convert_layouts to flash_attention.  Let me dig into that.

Reply:
I think I understand the problem and can at least avoid it for now, but it's worth digging into.  I left a TODO.